File: src/Models/utilisateur.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Models;



/**
 *
 * @author Mariem
 */
public class utilisateur {
    
    //joinTab = "SELECT *"   From user INNER JOIN avis ON user.cin = avis.cin ;  
 private int  id,cin;
 private String nom ,prenom ,mail,password ,role ;
private int  tel;
    public utilisateur() {
    }

    public utilisateur(int cin, String nom, String prenom,String mail, String password, String role, int tel) {
        this.cin = cin;
        this.nom = nom;
        this.prenom = prenom;
        this.mail= mail;
        this.password = password;
        this.role = role;
        this.tel = tel;
    }

    public utilisateur(int id, int cin, String nom, String prenom,String mail, String password, String role, int tel) {
        this.id = id;
        this.cin = cin;
        this.nom = nom;
        this.prenom = prenom;
        this.mail= mail;
        this.password = password;
        this.role = role;
        this.tel = tel;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getCin() {
        return cin;
    }

    public void setCin(int cin) {
        this.cin = cin;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getPrenom() {
        return prenom;
    }

    public void setPrenom(String prenom) {
        this.prenom = prenom;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }

    public int getTel() {
        return tel;
    }

    public void setTel(int tel) {
        this.tel = tel;
    }

    public String getMail() {
        return mail;
    }

    public void setMail(String mail) {
        this.mail = mail;
    }

    @Override
    public String toString() {
        return "utilisateur{" + "id=" + id + ", cin=" + cin + ", nom=" + nom + ", prenom=" + prenom + ", mail=" + mail + ", password=" + password + ", role=" + role + ", tel=" + tel + '}';
    }

    

    
    
    
}

File: src/Entity/Utilisateur.php
<?php

namespace App\Entity;

use App\Repository\UtilisateurRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Component\Validator\Constraints\Email;

#[ORM\Entity(repositoryClass: UtilisateurRepository::class)]
class Utilisateur
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column]
    /**
 *  @Assert\NotBlank(message="Le champ cin est obligatoire")
 *  @Assert\Regex(
 *     pattern="/^\d{8}$/",
 *     message="Le champ cin doit contenir 8 chiffres."
 * )
 */
    private ?int $cin = null;
    

    #[ORM\Column(length: 60)]
    /** 
    *
     * @Assert\Type("string",message="The value {{ value }} is not a valid {{ type }}.")
     *    @Assert\NotBlank(message="Nom taille entre [3..20]")
     *  @Assert\Length(
     *      min = 3,
     *      max = 20,
     *      minMessage = "le nom doit etre superieur a  3  characters ",
     *      maxMessage = "le nom doit etre inferieur a  20  characters"
     * )
     *  @Assert\Regex(
     *     pattern="/^[a-zA-Z]+$/",
     *     message="Ce champ ne doit contenir que des caractères alphabétiques."
     * )
     *  
    */
    private ?string $nom = null;

    #[ORM\Column(length: 80)]
    /** 
    *
     * @Assert\Type("string",message="The value {{ value }} is not a valid {{ type }}.")
     *    @Assert\NotBlank(message="Nom taille entre [3..20]")
     *  @Assert\Length(
     *      min = 3,
     *      max = 20,
     *      minMessage = "le nom doit etre superieur a  3  characters ",
     *      maxMessage = "le nom doit etre inferieur a  20  characters"
     * )
     *  @Assert\Regex(
     *     pattern="/^[a-zA-Z]+$/",
     *     message="Ce champ ne doit contenir que des caractères alphabétiques."
     * )
     *  
    */
    private ?string $prenom = null;

    #[ORM\Column(length: 200)]
    /**
 * @Assert\Email(
 *     message = "L'adresse e-mail '{{ value }}' n'est pas valide.",
 
 * )
 */

    private ?string $mail = null;

    #[ORM\Column(length: 220)]
      /**
 *  @Assert\NotBlank(message="Le champ password est obligatoire")
 * )
 */
    private ?string $password = null;

    #[ORM\Column(length: 210)]
      /**
 *  @Assert\NotBlank(message="Le champ password est obligatoire")
 * )
 */
    private ?string $role = null;

    #[ORM\Column]
     /**
 *  @Assert\NotBlank(message="Le champ role est obligatoire")
 * )
 */
    private ?int $tel = null;

    #[ORM\OneToMany(mappedBy: 'user', targetEntity: Avis::class)]
    private Collection $avis;

    public function __construct()
    {
        $this->avis = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getCin(): ?int
    {
        return $this->cin;
    }

    public function setCin(int $cin): self
    {
        $this->cin = $cin;

        return $this;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getPrenom(): ?string
    {
        return $this->prenom;
    }

    public function setPrenom(string $prenom): self
    {
        $this->prenom = $prenom;

        return $this;
    }

    public function getMail(): ?string
    {
        return $this->mail;
    }

    public function setMail(string $mail): self
    {
        $this->mail = $mail;

        return $this;
    }

    public function getPassword(): ?string
    {
        return $this->password;
    }

    public function setPassword(string $password): self
    {
        $this->password = $password;

        return $this;
    }

    public function getRole(): ?string
    {
        return $this->role;
    }

    public function setRole(string $role): self
    {
        $this->role = $role;

        return $this;
    }

    public function getTel(): ?int
    {
        return $this->tel;
    }

    public function setTel(int $tel): self
    {
        $this->tel = $tel;

        return $this;
    }

    /**
     * @return Collection<int, Avis>
     */
    public function getAvis(): Collection
    {
        return $this->avis;
    }

    public function addAvi(Avis $avi): self
    {
        if (!$this->avis->contains($avi)) {
            $this->avis->add($avi);
            $avi->setUser($this);
        }

        return $this;
    }

    public function removeAvi(Avis $avi): self
    {
        if ($this->avis->removeElement($avi)) {
            // set the owning side to null (unless already changed)
            if ($avi->getUser() === $this) {
                $avi->setUser(null);
            }
        }

        return $this;
    }
}
